mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndex.java

```diff
@@ -506,8 +518,21 @@ public Flux<MessageId> search(MailboxSession session, Collection<MailboxId> mail
     }
 
     private List<SearchResult> searchMultimap(Collection<MailboxId> mailboxIds, SearchQuery searchQuery) throws MailboxException {
-        ImmutableList.Builder<SearchResult> results = ImmutableList.builder();
+        return searchDocument(mailboxIds, searchQuery, maxQueryResults)
+            .stream()
+            .map(this::documentToSearchResult)
+            .toList();
+    }
+
+    private SearchResult documentToSearchResult(Document doc) {
+        MessageUid uid = MessageUid.of(doc.getField(UID_FIELD).numericValue().longValue());
+        MailboxId mailboxId = mailboxIdFactory.fromString(doc.get(MAILBOX_ID_FIELD));
+        Optional<MessageId> messageId = Optional.ofNullable(doc.get(MESSAGE_ID_FIELD))
+            .map(messageIdFactory::fromString);
+        return new SearchResult(messageId, mailboxId, uid);
+    }
 
+    public List<Document> searchDocument(Collection<MailboxId> mailboxIds, SearchQuery searchQuery, int maxQueryResults) throws MailboxException {
```

Comment:
I'm not sure, we should reuse this api (`LuceneMessageSearchIndex.searchDocument`) for `LuceneMemorySearchHighlighter` or not

I'm not yet understand how `org.apache.lucene.search.highlight.Highlighter` work, 
Will it  reuse the Document list, and hightlight based on that result.
Or it will search directly from Directory and result Document with hightlight...

Reply:
> Or it will search directly from Directory and result Document with highlight...

I think this one.

Reply:
it corrects